crates/cargo-contract/src/cmd/extrinsics/events.rs

```diff
@@ -19,100 +19,175 @@ use super::{
     DefaultConfig,
 };
 use crate::{
-    maybe_println,
     Verbosity,
     DEFAULT_KEY_COL_WIDTH,
 };
 use colored::Colorize as _;
 use transcode::{
     ContractMessageTranscoder,
     TranscoderBuilder,
+    Value,
 };
 
-use anyhow::Result;
+use anyhow::{
+    Ok,
+    Result,
+};
+use std::fmt::Write;
 use subxt::{
     self,
     events::StaticEvent,
     tx::TxEvents,
 };
 
-pub fn display_events(
-    result: &TxEvents<DefaultConfig>,
-    transcoder: &ContractMessageTranscoder,
-    subxt_metadata: &subxt::Metadata,
-    verbosity: &Verbosity,
-) -> Result<()> {
-    if matches!(verbosity, Verbosity::Quiet) {
-        return Ok(())
+/// Field that represent data of the event from contract call
+#[derive(serde::Serialize)]
+pub struct Field {
+    /// name of a field
+    pub name: String,
+    /// value of a field
+    pub value: Value,
+}
+
+impl Field {
+    pub fn new(name: String, value: Value) -> Self {
+        Field { name, value }
     }
+}
 
-    println!();
+/// Events produced from calling a contract
+#[derive(serde::Serialize)]
+pub struct Event {
+    /// name of a pallet
+    pub pallet: String,
+    /// name of the event
+    pub name: String,
+    /// data associated with the event
+    pub fields: Vec<Field>,
+}
 
-    if matches!(verbosity, Verbosity::Verbose) {
-        println!("VERBOSE")
-    }
+/// Result of the contract call
+#[derive(Default, serde::Serialize)]
+pub struct CallResult {
+    /// Instantiated contract hash
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub contract: Option<String>,
+    /// Instantiated code hash
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub code_hash: Option<String>,
+    /// The verbosity flags.
+    pub verbosity: Verbosity,
```

Comment:
I'm pretty sure we don't want to display the verbosity anyway, perhaps can be passed just to `display_events`